mlir/lib/Conversion/ArithToEmitC/ArithToEmitCPass.cpp

```diff
@@ -30,9 +30,42 @@ namespace {
 struct ConvertArithToEmitC
     : public impl::ConvertArithToEmitCBase<ConvertArithToEmitC> {
   void runOnOperation() override;
+
+  /// Applies conversion to opaque types for f80 and i80 types, both unsupported
+  /// in emitc. Used to test the pass with opaque types.
+  void populateOpaqueTypeConversions(TypeConverter &converter);
 };
 } // namespace
 
+void ConvertArithToEmitC::populateOpaqueTypeConversions(
```

Comment:
I don't understand why these types should be unconditionally legalized, and why only for bitwidth 80?